File: Android/app/src/main/java/io/agora/typing/base/Logger.kt
package io.agora.typing.base

import android.util.Log

enum class LogLevel {
    Info, Warning, Error
}

class Logger {
    companion object {
        const val debug = true
        const val tag = "chat"

        fun log(message: String, level: LogLevel) {
            if (!debug && level != LogLevel.Error) {
                return
            }
            when (level) {
                LogLevel.Info -> Log.d(tag, "$message (${Thread.currentThread().name})")
                LogLevel.Warning -> Log.w(tag, message)
                LogLevel.Error -> Log.e(tag, message)
            }
        }
    }
}

File: iOS/Typing/Base/Logger.swift
//
//  Logger.swift
//  OpenChat
//
//  Created by XC on 2021/1/16.
//

import Foundation

enum LogLevel {
    case info, warning, error
    
    var description: String {
        switch self {
        case .info:    return "Info"
        case .warning: return "Warning"
        case .error:   return "Error"
        }
    }
}

class Logger {
    
    fileprivate static let debug = true
    
    static func log(message: String, level: LogLevel) {
        if !debug && level != .error {
            return
        }
        print("\(level.description): \(message)")
    }
}
